dace/codegen/compiled_sdfg.py

```diff
@@ -273,6 +278,241 @@ def __call__(self, *args, **kwargs):
+                        compiledSdfg._lib.unload()
+                        raise
+                elif mode == 'load':
+                    # Loads a new SDFG if its different to the current one
+                    sdfg = vscode.stop_and_load()
+                    if (sdfg and
+                            sdfg.hash_sdfg() != compiledSdfg._sdfg.hash_sdfg()):
+                        compiledSdfg = sdfg.compile()
+                elif mode == 'save':
+                    # Saves the current SDFG to location specified by the user
+                    vscode.stop_and_save(compiledSdfg._sdfg)
+                elif mode == 'transform':
+                    # Renders the SDFG in VSCode and loads the new SDFG
+                    sdfg = vscode.stop_and_transform(compiledSdfg._sdfg)
+                    if sdfg.hash_sdfg() != compiledSdfg._sdfg.hash_sdfg():
+                        # Rename to mitigate the library LINK ERROR
+                        sdfg.name = sdfg.name + '_t'
+                        compiledSdfg._lib.unload()
+                        compiledSdfg = sdfg.compile()
+                elif mode == 'report' or (mode == 'verification'
+                                          and 'foldername' in response):
+                    # REPORT:   Create a report of the SDFG by dumping intermidiate
+                    #           results to an accuracy file
+                    # VERIFICATION: Creates a report with the same initial args
+                    #           as the referenced report. Computes the difference
+                    #           of the intermidiate results between both reports.
+                    if mode == 'verification':
+                        refReportFolder = response['foldername']
+                        # Check if the folder is a accuracy report folder
+                        is_report = True
+                        for filename in os.listdir(refReportFolder):
+                            if (not filename.endswith('.bin')
+                                    and not filename.endswith('.npy')):
+                                is_report = False
+                                break
+                        if not is_report:
+                            print('The folder is not a accuracy report folder')
+                            # Skip this run and give the user another chance to pick
+                            response = vscode.send_bp_recv(
+                                {'type': 'sdfgEditMode'})
+                            mode = response['mode']
+                            continue
+
+                    # Create a deep copy as we dont want to
+                    # modify the real sdfg
+                    sdfg = copy.deepcopy(compiledSdfg._sdfg)
+                    hasInstr = vscode.sdfg_has_instrumentation(sdfg)
+                    if hasInstr:
+                        vscode.sdfg_remove_instrumentations(sdfg)
+
+                    # Make every array persistent so that we can access the data
```

Comment:
I think you should refactor this code below to a separate function, just like you did with the other modes